src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Credential/CredentialService.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Performance.SDK;
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core.Credential
+{
+    public class CredentialService : ICredentialService
+    {
+        private readonly IEnumerable<ICredentialProvider> providers;
+
+        private static readonly Semaphore ProviderSemaphore = new Semaphore(1, 1);
+
+        public CredentialService(IEnumerable<ICredentialProvider> providers)
+        {
+            this.providers = providers;
+        }
+
+        public async Task<ICredentials> GetCredentialsAsync(Uri uri, CancellationToken cancellationToken)
+        {
+            Guard.NotNull(uri, nameof(uri));
+
+            ICredentials creds = null;
+
+            foreach (ICredentialProvider provider in this.providers)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+
+                    ProviderSemaphore.WaitOne();
+
+                    creds = await provider.GetAsync(uri, cancellationToken);
+                    if (creds != null)
+                    {
+                        break;
+                    }
+                }
```

Comment:
Without a catch block this will still throw if the code above throws. I assume you don't want that behavior

I'm not sure what this class is doing, though

Reply:
Removed for now